shared/js/background/classes/site.es6.js

```diff
@@ -1,29 +1,49 @@
 /**
- * Each Site creates its own Score instance. The attributes
- * of the Score are updated as we process new events e.g. trackers
+ * Each Site creates its own Grade instance. The attributes
+ * of the Grade are updated as we process new events e.g. trackers
  * blocked or https status.
  *
- * The Score attributes are then used generate a site
- * privacy score used in the popup.
+ * The Grade attributes are then used generate a site
+ * privacy grade used in the popup.
  */
 const settings = require('../settings.es6')
-const Score = require('./score.es6')
 const utils = require('../utils.es6')
 const abpLists = require('../abp-lists.es6')
+const privacyPractices = require('../privacy-practices.es6')
+const Grade = require('@duckduckgo/privacy-grade').Grade
+const trackerPrevalence = require('../../../data/tracker_lists/prevalence')
 
 class Site {
-    constructor (domain) {
-        if (domain) domain = domain.toLowerCase()
+    constructor (url) {
+        this.url = url || ''
+
+        let domain = utils.extractHostFromURL(this.url) || ''
+        domain = domain.toLowerCase()
+
         this.domain = domain
         this.trackerUrls = []
-        this.score = new Score(this.specialDomain(), this.domain)
+        this.grade = new Grade()
         this.whitelisted = false // user-whitelisted sites; applies to all privacy features
         this.setWhitelistStatusFromGlobal(domain)
         this.isBroken = this.checkBrokenSites(domain) // broken sites reported to github repo
         this.didIncrementCompaniesData = false
 
-        // set isSpecialDomain when the site is created. This value may be
-        // updated later by the onComplete listener
+        this.tosdr = privacyPractices.getTosdr(domain)
+
+        this.parentEntity = utils.findParent(domain) || ''
+        this.parentPrevalence = trackerPrevalence[this.parentEntity] || 0
+
+        if (this.parentEntity && this.parentPrevalence) {
+            this.grade.setParentEntity(this.parentEntity, this.parentPrevalence)
+        }
+
+        this.grade.setPrivacyScore(privacyPractices.getTosdrScore(domain))
+
+        if (this.url.match(/^https:\/\//)) {
+            this.grade.setHttps(true, true)
+        }
```

Comment:
@MariagraziaAlastra this resolves the delayed score bug. Basically, HTTPS on the grade didn't get updated until the tab had finished loading, so we considered HTTPS to be false until then.

Which was kind of silly because we'd already know the URL contains "https".

I've added an initial check to the `Site` class, and also left the final check when the tab finishes loading, just in case we ended up downgrading the tab.

(This also made this check on startup redundant: https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/290/files#diff-3cd94a6f60c9bfd33dffeca9dc0735f8L199)

Reply:
Hey @andrey-p, actually I see a slightly different behavior here now. Again on roam.co, now initially the original grade vs enhanced are B and B+. Then it gets updated to D and B+.
I wonder if there are other parts of the score that get loaded later on?

Reply:
I'll look at it. This may be due to more trackers being loaded as the page does its thing though.